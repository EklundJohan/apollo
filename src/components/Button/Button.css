.button {
  height: 48px;  
  width: 154px;
  margin: 5px 0px;
  font-size: 16px;
  border-radius: 6px;
  border: 1px solid var(--color-dark-gray);
  background-color: var(--color-white);
  color: var(--color-charcoal);
}

.button:hover {
  background-color: var(--color-marble);
}

.button:disabled {
  background-color: var(--color-marble);
  border-color: var(--color-marble);
}

.button-primary {
  background-color: var(--color-one-blue);
  border-color: var(--color-one-blue);
  color: var(--color-white);
}

.button-primary:hover,
.button-primary:disabled {
  background-color: var(--color-one-dark-blue);
}

.button-danger { 
  /* height: 48px; */
  /* widows: 150px; */
  background-color: var(--color-red-status);
  border-color: var(--color-red-status);
  color: var(--color-white);
}

.button-danger:hover,
.button-danger:disabled {
  background-color: var(--color-red-hover-status);
}

.button-success {
  /* height: 48px; */
  /* width: 162px; */
  background-color: var(--color-green-status);
  border-color: var(--color-green-status);
  color: var(--color-white);
}

.button-success:hover,
.button-success:disabled {
  background-color: var(--color-green-hover-status);
}

.button-small {
  height: 40px;
  width: 144px;
}

.button-large {
  height: 56px;
  width: 145px;
}
