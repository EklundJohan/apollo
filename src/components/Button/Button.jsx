import React from 'react';
import PropTypes from 'prop-types';

import * as styles from './Button.css';

export default class Button extends React.Component {
    render() {
        return (
            <button 
                onClick={this.props.onClick}
                onMouseEnter={this.props.onMouseEnter}
                onMouseLeave={this.props.onMouseLeave}
                disabled={this.props.disabled}>
                {this.props.children}</button>
        );
    };
};

Button.defaultProps = {

};

Button.propTypes = {
    children: PropTypes.string,
    disabled: PropTypes.bool,
    onClick: PropTypes.func,
    onMouseEnter: PropTypes.func,
    onMouseLeave: PropTypes.func
};
