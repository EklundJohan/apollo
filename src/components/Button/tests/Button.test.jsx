import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe('<Button />', () => {
    describe('onClick()', () => {
        it('is called when the button is clicked', () => {
            const spy = jest.fn();
            renderer.create(<Button onClick={spy} />).toJSON().props.onClick();
            expect(spy).toHaveBeenCalled();
        });
    });

    describe('onFocus()', () => {
        it('is called when the button is focused', () => {
            const spy = jest.fn();
            renderer.create(<Button onFocus={spy} />).toJSON().props.onFocus();
            expect(spy).toHaveBeenCalled();
        });
    });

    describe('onBlur()', () => {
        it('is called when the button is blurred', () => {
            const spy = jest.fn();
            renderer.create(<Button onBlur={spy} />).toJSON().props.onBlur();
            expect(spy).toHaveBeenCalled();
        });
    });

    describe('disabled', () => {
        it('sets the disabled attribute on the button', () => {
            const button = renderer.create(<Button disabled />);
            expect(button.toJSON().props.disabled).toBe(true);
        });

        it('sets the disabled attribute on the button', () => {
            const button = renderer.create(<Button disabled={false} />);
            expect(button.toJSON().props.disabled).toBe(false);
        });
    });

    describe('type', () => {
        it('sets the type attribute on the button to default', () => {
            const button = renderer.create(<Button />);
            expect(button.toJSON().props.type).toBe('button');
        });

        it('sets the type attribute on the button to submit', () => {
            const button = renderer.create(<Button type='submit' />);
            expect(button.toJSON().props.type).toBe('submit');
        });

        it('sets the type attribute on the button to reset', () => {
            const button = renderer.create(<Button type='reset' />);
            expect(button.toJSON().props.type).toBe('reset');
        });
    });
});
