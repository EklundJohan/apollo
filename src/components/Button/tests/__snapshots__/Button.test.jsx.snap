// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`onClick 1`] = `
<button
  disabled={undefined}
  onClick={[Function]}
  onMouseEnter={undefined}
  onMouseLeave={undefined}
/>
`;

exports[`onMouseEnter 1`] = `
<button
  disabled={undefined}
  onClick={undefined}
  onMouseEnter={[Function]}
  onMouseLeave={undefined}
/>
`;

exports[`onMouseLeave 1`] = `
<button
  disabled={undefined}
  onClick={undefined}
  onMouseEnter={undefined}
  onMouseLeave={[Function]}
/>
`;
